experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/types.ts

```diff
@@ -3,16 +3,33 @@
  * Licensed under the MIT License.
  */
 
-import { AllowedTypes } from "../../typed-schema";
+import { SchemaAware } from "../..";
+import { RestrictiveReadonlyRecord } from "../../../util";
+import { FieldKinds } from "../../default-field-kinds";
+import { FieldKind } from "../../modular-schema";
+import {
+	AllowedTypes,
+	Any,
+	FieldNodeSchema,
+	FieldSchema,
+	InternalTypedSchemaTypes,
+	LeafSchema,
+	MapSchema,
+	StructSchema,
+	TreeSchema,
+} from "../../typed-schema";
 import {
 	UnboxNodeUnion,
 	CheckTypesOverlap,
 	FlexibleNodeContent,
 	Sequence,
+	NodeKeyField,
+	AssignableFieldKinds,
 } from "../editableTreeTypes";
 
 /** Implements 'readonly T[]' and the list mutation APIs. */
-export interface List<TTypes extends AllowedTypes> extends ReadonlyArray<UnboxNodeUnion<TTypes>> {
+export interface SharedTreeList<TTypes extends AllowedTypes>
```

Comment:
Why the rename? Do we plan on doing the same thing with map?

Reply:
I see we did. Out of curiosity, why?

Reply:
Yes. `SharedTreeObject` and `SharedTreeMap` are just below. "List" is okay, but we can't call the object "Object" and the map "Map".